parseAndPopulate/groupings.py

```diff
@@ -96,6 +100,81 @@ def parse_and_load(self):
+        elif 'sysrepo' in split_path:
+            if 'internal' in split_path:
+                return 'SysrepoInternal'
+            elif 'applications' in split_path:
+                return 'SysrepoApplication'
+        if 'tmp' in split_path:
+            if schema_base in self._submodule_map:
+                split_path = self._submodule_map[schema_base].split('/') + split_path[4:]
+        else:
+            split_path = split_path[split_path.index('yang', 3) + 1:]
+        module_category, organization, split_path = split_path[0], split_path[1], split_path[2:]
+        if module_category == 'standard':
+            if organization == 'ietf':
+                module_type = split_path[0]
+                if module_type == 'DRAFT':
+                    return 'IETFDraft'
+                elif module_type == 'RFC':
+                    return 'IETFYANGRFC'
+            elif organization == 'etsi':
+                if split_path[0] == 'NFV-SOL006-v2.6.1':
+                    return 'ETSI261'
+                elif split_path[0] == 'NFV-SOL006-v2.7.1':
+                    return 'ETSI271'
+            elif organization == 'bff':
+                return 'BFF'
+            elif organization == 'mef':
+                if 'standard' in split_path[-2:]:
+                    return 'MEFStandard'
+                elif 'draft' in split_path[-2:]:
+                    return 'MEFExperimental'
+            elif organization == 'ieee':
+                module_status = split_path[0]
+                if module_status == 'published':
+                    return 'IEEEStandard'
+                elif module_status == 'draft':
+                    return 'IEEEStandardDraft'
+            elif organization == 'iana':
+                return 'IANAStandard'
+        elif module_category == 'experimental':
+            if organization == 'ieee':
+                return 'IEEEExperimental'
+        elif module_category == 'vendor':
+            if organization == 'cisco':
+                cisco_os, version = split_path[0], split_path[1]
+                return 'Cisco{}{}'.format(cisco_os.upper(), alnum(version))
+            elif organization == 'juniper':
+                version = split_path[0]
+                if version == '14.2':
+                    return 'Juniper142'
+                else:
+                    version_detail = split_path[1]
+                    return 'Juniper{}'.format(alnum(version_detail))
+            elif organization == 'huawei':
+                version  = split_path[1]
+                return 'NETWORKROUTER{}'.format(alnum(version))
+            elif organization == 'ciena':
+                return 'CIENA'
+            elif organization == 'fujitsu':
+                return 'Fujitsu{}{}'.format(alnum(split_path[1]), alnum(split_path[2]))
+            elif organization == 'nokia':
+                version = split_path[1].removeprefix('latest_sros_')
+                return 'Nokia{}'.format(alnum(version))
+        return None
```

Comment:
Is it not possible to create some kind of nested dictionary that will contain info for each repo? 
It would be stored in staticVaribles.py and this dictionary would be modified instead of a large number of if-else blocks.

Reply:
It would be possible for a few, but for most the process is more complicated than just a search and replace. The only way I see is to make a dictionary with lambda functions, which I think is less clear. The script that creates the files also handles them case by case.

The other option would be to change the way the files are named when they are created. I'd have to check what else depends on the names of the JSONs and HTMLs and it might require wider changes.